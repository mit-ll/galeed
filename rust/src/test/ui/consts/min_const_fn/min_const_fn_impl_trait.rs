struct AlanTuring<T>(T);
const fn no_rpit2() -> AlanTuring<impl std::fmt::Debug> {
    //~^ ERROR `impl Trait` in const fn is unstable
    AlanTuring(0)
}

const fn no_rpit() -> impl std::fmt::Debug {} //~ ERROR `impl Trait` in const fn is unstable

fn main() {}
