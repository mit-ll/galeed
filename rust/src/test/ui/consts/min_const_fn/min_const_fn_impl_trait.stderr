error[E0723]: `impl Trait` in const fn is unstable
  --> $DIR/min_const_fn_impl_trait.rs:2:24
   |
LL | const fn no_rpit2() -> AlanTuring<impl std::fmt::Debug> {
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error[E0723]: `impl Trait` in const fn is unstable
  --> $DIR/min_const_fn_impl_trait.rs:7:23
   |
LL | const fn no_rpit() -> impl std::fmt::Debug {}
   |                       ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0723`.
